e2e/machineset_migration_test.go

```diff
@@ -213,13 +214,676 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 			})
 
 			It("should verify that CAPI MachineSet has Paused condition False", func() {
-				verifyCAPIPausedCondition(capiMachineSet, machinev1beta1.MachineAuthorityClusterAPI)
+				verifyCAPIPausedCondition(capiMachineSet, mapiv1beta1.MachineAuthorityClusterAPI)
+			})
+		})
+	})
+
+	var _ = Describe("Scale MAPI MachineSets", Ordered, func() {
+		var mapiMSAuthMAPIName = "ms-authoritativeapi-mapi"
+		var mapiMSAuthCAPIName = "ms-authoritativeapi-capi"
+		var mapiMSAuthMAPICAPI = "ms-mapi-machine-capi"
+
+		var awsMachineTemplate *capav1.AWSMachineTemplate
+		var capiMachineSet *capiv1beta1.MachineSet
+		var mapiMachineSet *mapiv1beta1.MachineSet
+		var firstMAPIMachine *mapiv1beta1.Machine
+		var secondMAPIMachine *mapiv1beta1.Machine
+		var err error
+
+		Context("with spec.authoritativeAPI: MAPI", Ordered, func() {
+			BeforeAll(func() {
+				mapiMachineSet, err = createMAPIMachineSetWithAuthoritativeAPI(ctx, cl, 1, mapiMSAuthMAPIName, mapiv1beta1.MachineAuthorityMachineAPI, mapiv1beta1.MachineAuthorityMachineAPI)
+				Expect(err).ToNot(HaveOccurred(), "MAPI MachineSet creation should succeed")
+
+				capiMachineSet, awsMachineTemplate = verifyMAPIMachineSetHasCAPIMirror(cl, mapiMSAuthMAPIName)
+
+				mapiMachines, err := mapiframework.GetMachinesFromMachineSet(ctx, cl, mapiMachineSet)
+				Expect(err).ToNot(HaveOccurred(), "failed to get MAPI Machines from MachineSet")
+				capiMachines, err := capiframework.GetMachinesFromMachineSet(cl, capiMachineSet)
+				Expect(err).ToNot(HaveOccurred(), "failed to get CAPI Machines from MachineSet")
+				Expect(capiMachines[0].Name).To(Equal(mapiMachines[0].Name))
+				firstMAPIMachine = mapiMachines[0]
+
+				DeferCleanup(func() {
+					By("Cleaning up Context 'with spec.authoritativeAPI: MAPI' resources")
+					cleanupTestResources(
+						ctx,
+						cl,
+						[]*capiv1beta1.MachineSet{capiMachineSet},
+						[]*capav1.AWSMachineTemplate{awsMachineTemplate},
+						[]*mapiv1beta1.MachineSet{mapiMachineSet},
+					)
+				})
+			})
+
+			It("should scale MAPI MachineSet to 2 replicas succeed", func() {
+				By("Scaling up MAPI MachineSet to 2 replicas")
+				Expect(mapiframework.ScaleMachineSet(mapiMachineSet.GetName(), 2)).To(Succeed(), "should be able to scale up MAPI MachineSet")
+				mapiframework.WaitForMachineSet(ctx, cl, mapiMSAuthMAPIName)
+				Eventually(komega.Object(mapiMachineSet)).Should(HaveField("Spec.Replicas", HaveValue(Equal(int32(2)))))
+
+				By("Verifying a new MAPI Machine is created and Paused condition is False")
+				Eventually(func() (*mapiv1beta1.Machine, error) {
+					secondMAPIMachine, err = mapiframework.GetLatestMachineFromMachineSet(ctx, cl, mapiMachineSet)
+					if err != nil {
+						return nil, err
```

Comment:
If this error throws, what do we log? 

We cold add additional context too: 


```suggestion
						return nil, err
						return nil, fmt.Errorf("failed to get latest machine from machineset %s: %s", mapiMachineSet.GetName(), err.Error())
```